backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/CustomTypeMapper.kt

```diff
@@ -74,6 +75,7 @@ internal object CustomTypeMappers {
 
     private fun ClassDescriptor.isMappedFunctionClass() =
             this.getFunctionalClassKind() == FunctionClassDescriptor.Kind.Function
+                    && (declaredTypeParameters.size - 1 < ObjCExportBlockCodeGenerator.functionTypeConvertorsArityLimit)// Type parameters include return type.
```

Comment:
I believe we can (and should!) move all comparisons like `arity <  functionTypeConvertorsArityLimit` to the single function. Otherwise we have this logic duplicated, which might be a source of undetectable errors.